pkg/loki/modules.go

```diff
@@ -766,11 +766,15 @@ func (t *Loki) initCompactor() (services.Service, error) {
 
 	t.Server.HTTP.Path("/compactor/ring").Methods("GET", "POST").Handler(t.compactor)
 
-	// TODO: update this when the other deletion modes are available
-	if t.Cfg.CompactorConfig.RetentionEnabled && t.compactor.DeleteMode() == deletion.WholeStreamDeletion {
-		t.Server.HTTP.Path("/loki/api/v1/delete").Methods("PUT", "POST").Handler(t.HTTPAuthMiddleware.Wrap(http.HandlerFunc(t.compactor.DeleteRequestsHandler.AddDeleteRequestHandler)))
-		t.Server.HTTP.Path("/loki/api/v1/delete").Methods("GET").Handler(t.HTTPAuthMiddleware.Wrap(http.HandlerFunc(t.compactor.DeleteRequestsHandler.GetAllDeleteRequestsHandler)))
-		t.Server.HTTP.Path("/loki/api/v1/delete").Methods("DELETE").Handler(t.HTTPAuthMiddleware.Wrap(http.HandlerFunc(t.compactor.DeleteRequestsHandler.CancelDeleteRequestHandler)))
+	if t.Cfg.CompactorConfig.RetentionEnabled {
```

Comment:
Do we still want these endpoints if the mode is `disabled`?

Reply:
I don't really see the point. If you can't delete why provide a way to add delete requests?

Reply:
That makes sense. Do we ever expect both retention-enabled to be true and delete-mode to be `disabled`?

Reply:
The way I understand https://grafana.com/docs/loki/latest/operations/storage/retention/ is that retention removes older data based on the `retention_period` setting. This is separate from explicit delete requests. So the scenario you mention is valid. @sandeepsukhani is that correct?

Reply:
Cool deal. That means we want our condition to be 

```go
if t.Cfg.CompactorConfig.RetentionEnabled && deleteMode != Disabled
```

Reply:
Yes, but you could argue that the current way is more future proof. If we add a new deletion mode that doesn't use the API this won't have to change. That is probably not a likely scenario though.

Reply:
I must be missing something here. If retention is enabled and delete mode = Disabled then the default branch in the switch statement is taken and the API is not added using `t.Server.HTTP.Path(...)`.